pkg/ruler/evaluator_remote.go

```diff
@@ -161,36 +227,59 @@ func (q *remoteQuerier) query(ctx context.Context, query string, ts time.Time, l
 			{Key: textproto.CanonicalMIMEHeaderKey("User-Agent"), Values: []string{userAgent}},
 			{Key: textproto.CanonicalMIMEHeaderKey("Content-Type"), Values: []string{mimeTypeFormPost}},
 			{Key: textproto.CanonicalMIMEHeaderKey("Content-Length"), Values: []string{strconv.Itoa(len(body))}},
+			{Key: textproto.CanonicalMIMEHeaderKey(string(httpreq.QueryTagsHTTPHeader)), Values: []string{"ruler"}},
+			{Key: textproto.CanonicalMIMEHeaderKey(user.OrgIDHeaderName), Values: []string{orgID}},
 		},
 	}
 
-	for _, mdw := range q.middlewares {
-		if err := mdw(ctx, &req); err != nil {
-			return nil, err
-		}
+	start := time.Now()
+	resp, err := r.client.Handle(ctx, &req)
+
+	instrument.ObserveWithExemplar(ctx, r.metrics.reqDurationSecs.WithLabelValues(orgID), time.Since(start).Seconds())
+
+	if resp != nil {
+		instrument.ObserveWithExemplar(ctx, r.metrics.responseSizeBytes.WithLabelValues(orgID), float64(len(resp.Body)))
 	}
 
-	start := time.Now()
-	resp, err := q.client.Handle(ctx, &req)
+	log := log.With(logger, "query_hash", hash, "query", query, "instant", ts, "response_time", time.Since(start).String())
+
 	if err != nil {
-		level.Warn(logger).Log("msg", "failed to remotely evaluate query expression", "err", err, "query_hash", hash, "qs", query, "ts", ts, "response_time", time.Since(start).Seconds())
-		return nil, err
+		r.metrics.failedEvals.WithLabelValues("error", orgID).Inc()
+
+		level.Warn(log).Log("msg", "failed to evaluate rule", "err", err)
+		return nil, fmt.Errorf("rule evaluation failed: %w", err)
 	}
+
+	// TODO(dannyk): consider retrying if the rule has a very high interval, or the rule is very sensitive to missing samples
+	//   i.e. critical alerts or recording rules producing crucial RemoteEvaluatorMetrics series
 	if resp.Code/100 != 2 {
-		return nil, fmt.Errorf("unexpected response status code %d: %s", resp.Code, string(resp.Body))
+		r.metrics.failedEvals.WithLabelValues("upstream_error", orgID).Inc()
```

Comment:
nit: Do you think better to put all these reasons (`error`, `upstream_error`, `max_size`, `timeout`) for failed evaluations in one place as package level variables? 

Also these reasons looks so generic? may be tweak a bit? particularly `error`.

Reply:
> nit: Do you think better to put all these reasons (error, upstream_error, max_size, timeout) for failed evaluations in one place as package level variables?

For what purpose? And I assume you mean constants?

> Also these reasons looks so generic? may be tweak a bit? particularly error.

Correct, we can't have too much cardinality in these labels and also there could be a very large number of possible failure cases. The logs are extensive, so the operator would use the logs to figure out what the actual issue was.

Reply:
> For what purpose? And I assume you mean constants?

Yep constants. sorry!.

Reply:
I don't really see the point of having them as package-level constants since their use is localized to this file and there is no duplication.